ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,220 @@
+  sendCountIsTrackable,
+  segment,
+  METAMETRICS_ANONYMOUS_ID,
+} from '../helpers/utils/metametrics.util'
+import { PATH_NAME_MAP } from '../helpers/constants/routes'
+import { getCurrentLocale } from '../ducks/metamask/metamask'
+import { txDataSelector } from '../selectors'
+
+export const MetaMetricsContext = createContext(() => {
+  captureException(
+    Error(`MetaMetrics context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
+  )
+})
+
+const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
+
+function useSegmentContext () {
+  const match = useRouteMatch({ path: PATHS_TO_CHECK, exact: true, strict: true })
+  const locale = useSelector(getCurrentLocale)
+  const txData = useSelector(txDataSelector) || {}
+  const confirmTransactionOrigin = txData.origin
+
+  const referrer = confirmTransactionOrigin ? {
+    url: confirmTransactionOrigin,
+  } : undefined
+
+  let version = global.platform.getVersion()
+  if (process.env.METAMASK_ENVIRONMENT !== 'production') {
+    version = `${version}-${process.env.METAMASK_ENVIRONMENT}`
+  }
+
+  const page = match ? {
+    path: match.path,
+    title: PATH_NAME_MAP[match.path],
+    url: match.path,
+  } : undefined
+
+  return {
+    app: {
+      version,
+      name: 'MetaMask Extension',
+    },
+    locale,
+    page,
+    referrer,
+    userAgent: window.navigator.userAgent,
+  }
+}
+
+export function MetaMetricsProvider ({ children }) {
+  const network = useSelector(getCurrentNetworkId)
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const location = useLocation()
+  const context = useSegmentContext()
+
+  // Used to prevent double tracking page calls
+  const previousMatch = useRef()
+
+  /**
+   * Anytime the location changes, track a page change with segment.
+   * Previously we would manually track changes to history and keep a
+   * reference to the previous url, but with page tracking we can see
+   * which page the user is on and their navigation path.
+   */
+  useEffect(() => {
+    const environmentType = getEnvironmentType()
+    if (location.pathname.startsWith('/initialize') || participateInMetaMetrics) {
+      // Events that happen during initialization before the user opts into MetaMetrics will be anonymous
+      const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
+      const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
+      const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
+      if (
+        match &&
+        previousMatch.current !== match.path &&
+        // If we're in a popup or notification we don't want the initial home route to track
+        ((environmentType !== 'popup' && environmentType !== 'notification') || match.path !== '/')
```

Comment:
We do want _subsequent_ navigations back to home to track though :thinking: Or at least I'd think we would. This seems to skip tracking _any_ navigations to home.

Would this still serve the purpose you had in mind?

```suggestion
        // If we're in a popup or notification we don't want the initial home route to track
        !(
          (environmentType === 'popup' || environmentType === 'notification')
          && match.path === '/'
          && previousMatch.current === undefined
        )
```